RADAR-base/radar-commons-android
# Refactor commons-android

- Converted Java code to Kotlin
- Updated dependencies
- Converted to `org.radarbase` package base.
- Added `org.radarbase.android.util.SafeHandler` to avoid common Handler pitfalls.
- Converted to `android.support` to `androidx`
- Run a single `AuthService` to keep the `AppAuthState`. This means that once services are running and bound to AuthService, they can always refresh the access token, whether the app is in the foreground or not.
- Added support for multiple sources with the same type.
- Use androidTest instead of mocking all Android types
- No longer include `sent` messages in `DataCache.numberOfRecords`
- Make call to `AbstractDeviceManager.register()` mandatory
- Configure device manager via `DeviceService.configureManager` instead of doing it three times via `DeviceServiceProvider`, `DeviceService.onInvocation` and `DeviceService.createDeviceManager`.
- Replace `AbstractDeviceManager.createTopic` with `AbstractDeviceManager.createCache`.
- Replace calls to `RadarConfiguration.getString(key, null)` with `RadarConfiguration.optString(key)`.
- Merge all plugins to a single directory for more efficient and reliable release management.
  - Missing plugin: Biovotion.
